crates/shared/src/token_list.rs

```diff
@@ -18,31 +51,46 @@ pub struct Token {
 }
 
 impl TokenList {
-    pub async fn from_url(url: impl IntoUrl, chain_id: u64, client: Client) -> Result<Self> {
-        let model: TokenListModel = client.get(url).send().await?.json().await?;
-        Ok(Self::from_tokens(model.tokens, chain_id))
-    }
+    pub async fn from_configuration(configuration: TokenListConfiguration) -> Self {
+        let tokens = Arc::new(RwLock::new(
+            configuration.tokens().await.unwrap_or_default(),
+        ));
```

Comment:
Do we really want unwrap_or_default()? That means on start up this can fail and we run without a token list. Not saying we have to change it just making sure that this is a conscious decision.

Reply:
I think this makes sense in that:
- We then disallow all internal tokens
- We spawn a background thread that updates the token lists and would eventually populate it with some values

I don't like that its silent though... Maybe we can add a log that warns when token list initialization fails.

Reply:
As @nlordell explained this was intentional. But I added log (error) for just in case.